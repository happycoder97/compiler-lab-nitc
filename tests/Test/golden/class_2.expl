class A         
{
  str f0() {   
       write("IN A");
       return "RET A";
  }
}

class B
{
  str f0() {   
       write("IN B");
       return "RET B";
  }
}

A a;
B b;

int main() {
    initialize();
    a = new(A);
    b = new(B);
    write(a.f0());
    write(b.f0());
}
