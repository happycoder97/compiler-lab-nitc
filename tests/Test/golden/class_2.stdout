IN A
RET A
IN B
RET B
